### ISSUE TITLE
Testing for 3:1 ratio between link text (or other valid indication) and it's surrounding text

### ISSUE THREAD

**Issue by joelhsmith** [dequelabs/axe-core#593]
I know there is already a rule looking for color contrast issues. Can we add a test for 3:1 ratio between link text (or other valid indication) and surrounding body text To meet [Success Criterion 1.4.1 (Use of Color)](https://www.w3.org/TR/2008/NOTE-WCAG20-TECHS-20081211/G183) The primary target I was hoping to catch is text in the body. But it would apply to everything I guess. 

This is my rough idea is:

**Pass:**

Link and its surrounding text have higher than 3:1 contrast ratio.
Or
Link has a property of `text-decoration` value of `underline`.
Or 
Link has a property of `border` with a value of not `none`.
Or
Link has a property of `background` or `background-color` or `background-image` with a value of not `none`.

**Fail:**

Link and its surrounding text have less than 3:1 contrast ratio 
AND
There is no property for `text-decoration` of `underline`.
AND 
There is no property of `border`.
AND
There is no property for `background` and `background-color` and `background-image`.

I don't think it should be a fail, but it probably should be flagged as `needs review`.

**Comment by WilcoFiers** [comment]
@joelhsmith Axe-core looks at the font size and font weight. Based on that it will either use 4.5:1 or 3:1. Did you have a specific case where that didn't happen? If so that is a bug.

**Comment by joelhsmith** [comment]
Whoops, it caught it. My bad. Closing.
